core/src/main/java/org/elasticsearch/script/LeafSearchScript.java

```diff
@@ -44,8 +56,23 @@ default void setNextAggregationValue(Object value) {
         setNextVar("_value", value);
     }
 
-    long runAsLong();
+    @Override
+    default void setNextVar(String field, Object value) {}
 
-    double runAsDouble();
+    /**
+     * Return the result as a long. This is used by aggregation scripts over long fields.
+     */
+    default long runAsLong() {
+        return (long) runAsDouble();
```

Comment:
I agree. I changed runAsLong to default to `UnsupportedOperationException` and tweaked the docs a tiny bit. Also note that with contexts coming this will all change significantly, as each context will have its own dedicated interface, so no more of this shared interface trying to satisfy multiple use cases.